Assets/PatchKit Patcher/Scripts/AppData/FileSystem/RetryStrategy.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using PatchKit.Network;
+using PatchKit.Unity.Patcher.Cancellation;
+using PatchKit.Unity.Patcher.Debug;
+using PatchKit.Unity.Utilities;
+
+namespace PatchKit.Unity.Patcher.AppData.FileSystem
+{
+    public class RetryStrategy : IRequestRetryStrategy
+    {
+        private static readonly DebugLogger DebugLogger = new DebugLogger(typeof(RetryStrategy));
+
+        public const int DefaultDuration = 5 * 1000;
+        public const int DefaultDelay = 1000 / 2;
+
+        private readonly int _tryCount;
+        private int _currentTry = 0;
+        private readonly int _delay;
+
+        private RetryStrategy()
+            : this(DefaultDuration, DefaultDelay)
+        {
+        }
+
+        public RetryStrategy(int tryDurationMsec, int delayBetweenEachTryMsec)
+        {
+            _tryCount = tryDurationMsec / delayBetweenEachTryMsec;
```

Comment:
Why calculating the amount of retries is based on duration and delay?

Reply:
I can add a constructor that would allow to specify the amount of retries and the delay between each of them. 

In this particular situation I decided to use the total duration and delay between each retry, because it felt more comfortable.

Reply:
Hm, I'd prefer simple constructor that takes amount of retries and delay between them. We don't need to complicate things here.